src/state/mod.rs

```diff
@@ -149,13 +156,97 @@ impl State {
                     b: x * y / (self.size.width as f64 * self.size.height as f64),
                     a: 1.0,
                 };
+                // self.clear_color = wgpu::Color {
+                //     r: 0.1,
+                //     g: 0.2,
+                //     b: 0.3,
+                //     a: 1.0,
+                // };
                 true
             }
             _ => false,
         }
     }
 
     pub fn update(&mut self) {
+        let angle = std::f32::consts::PI;
+        let new_vertex = Vertex::rotate_vertex(angle, Vertex::VERTICES_PENTAGON);
+        // let num_triangles = self.num_vertices - 2;
+        // let new_indices = (1u16..num_triangles as u16 + 1)
+        //     .into_iter()
+        //     .flat_map(|i| vec![i + 1, i, 0])
+        //     .collect::<Vec<_>>();
+        let new_indices = Vertex::INDICES;
+
+        let cartoon_bytes = include_bytes!("happy-tree-cartoon.png");
+        let cartoon_texture = Texture::from_bytes(
+            &self.device,
+            &self.queue,
+            cartoon_bytes,
+            "happy-tree-cartoon",
+        ).unwrap();
+
+        match self.use_color {
+            true => {
```

Comment:
The expression inside true and false seems to share many similarities. Maybe it needs to be inside a function

Reply:
Refactored.
```rust
        match self.bool_event {
            true => {
                self.vertex_buffer = Buffer::new(
                    &self.device,
                    Vertex::VERTICES_PENTAGON,
                    wgpu::BufferUsages::VERTEX,
                    "Vertex Buffer",
                );
                self.index_buffer = Some(Buffer::new(
                    &self.device,
                    Vertex::INDICES,
                    wgpu::BufferUsages::INDEX,
                    "Index Buffer",
                ));
                self.num_indices = Vertex::INDICES.len() as u32;
                self.texture_bind_group = Texture::init_texture_bind_group(
                    &self.device,
                    &self.default_texture,
                    &self.texture_bind_group_layout,
                    "diffuse_texture_bind_group",
                );
            }
            false => {
                self.vertex_buffer = Buffer::new(
                    &self.device,
                    &new_vertex,
                    wgpu::BufferUsages::VERTEX,
                    "Challenge Vertex Buffer",
                );
                self.index_buffer = Some(Buffer::new(
                    &self.device,
                    &new_indices,
                    wgpu::BufferUsages::INDEX,
                    "Index Buffer",
                ));
                self.num_indices = new_indices.len() as u32;
                self.texture_bind_group = Texture::init_texture_bind_group(
                    &self.device,
                    &self.cartoon_texture,
                    &self.texture_bind_group_layout,
                    "cartoon_texture_bind_group",
                );
            }
        };
```